### ISSUE TITLE
Debian12 Portmaster failed to start module

### ISSUE THREAD

**Issue by LiberteIV** [safing/portmaster#1239]
**Pre-Submit Checklist**:

- Check applicable sources for existing issues:
  - [Windows Known Issues](https://docs.safing.io/portmaster/install/windows#known-issues)
  - [Linux Known Issues](https://docs.safing.io/portmaster/install/linux#compatibility)
  - [Github Issues](https://github.com/safing/portmaster/issues?q=is%3Aissue+label%3Abug)

**What happened**:
installed portmaster via the .deb installer from safing.io
debian version is bookworm (12)
error code: Failed to start module: could not initialize nfqueue: exec: "iptables":executable file not found in $PATH
has the safing team tested portmaster on debian12? why does this error occur? how to resolve the issue?
![0](https://github.com/safing/portmaster/assets/116210112/6055173f-84b7-4909-ac51-e27f0bf400f7)

**What did you expect to happen?**:

**How did you reproduce it?**:

**Debug Information**:

**Comment by LiberteIV** [comment]
According to a member of the debian forums:

![1](https://github.com/safing/portmaster/assets/116210112/26b55ef7-e261-40ff-b430-f2e3de921914)

![2](https://github.com/safing/portmaster/assets/116210112/361e77c7-d413-4629-93ff-e0ca3ee69577)

https://wiki.debian.org/nftables

Please note these aren't my words and i'm not here to upset anyone, i'm sharing information.

**Comment by LiberteIV** [comment]
Anyone??? no reply after 3 days wth

**Comment by Raphty** [comment]
Hey there @LiberteIV,
no, none of us are using Debian on a daily basis.

and to be honest so view people do, that I am not sure when we will get around fixing this, since the impact is low.

if someone in the community wants to fix it and give a PR let us know in advance and lets coordinate this.

**Comment by LiberteIV** [comment]
> Hey there @LiberteIV, no, none of us are using Debian on a daily basis.
> 
> and to be honest so view people do, that I am not sure when we will get around fixing this, since the impact is low.
> 
> if someone in the community wants to fix it and give a PR let us know in advance and lets coordinate this.

If you are not using debian, what linux distro do you recommend then?
Don't get me wrong, i'm not asking for linux distro suggestions as that would be the wrong place.
But which distro has the best portmaster support? Windows sucks.

**Comment by Morasithil** [comment]
> Hey there @LiberteIV, no, none of us are using Debian on a daily basis.
> 
> and to be honest so view people do, that I am not sure when we will get around fixing this, since the impact is low.
> 
> if someone in the community wants to fix it and give a PR let us know in advance and lets coordinate this.

I wouldn't say that view people use debian, it's one of the oldest linux distros out there and it is popular.
If you are not going to fix your portmaster deb installer then you should remove it from your website.
You are offering a download for a product that does not work, it makes no sense.

**Comment by Morasithil** [comment]
Plus, this is NOT net info.
NOTE: iptables was replaced by [nftables](https://wiki.debian.org/nftables) starting in Debian 10 Buster.
We have debian 12 now...
https://wiki.debian.org/iptables

**Comment by LiberteIV** [comment]
This

**Comment by LiberteIV** [comment]
what a joke

**Comment by Raphty** [comment]
thanks for the report, we see that some people are complaining about issues on debian.

If you want to help that would be great, we can't and will not divert resources into this at the moment.

to address the .deb installer, many distros based on ubuntu use it, and Portmaster works well on those. It also seems that Portmaster worked well on deb11, as mentioned we currently don't have the resources to divert to fixing deb12, but your help and contribution to bringing Portmaster to deb12 would be super welcome!

**Comment by mattkeenan** [comment]
Hi Raphty,

I can have a look at doing this, if you want. I've never used portmaster before but it looks interesting so it may take me a bit to get up to speed, but I'd love to help.

**Comment by Raphty** [comment]
Thanks @mattkeenan , @dhaavi is coordinating such efforts. But if you have question I gladly help where I can 😄

**Comment by mattkeenan** [comment]
Since this very much looks like a bug in safing/portmaster-packaging#89 maybe this should be closed? My fix / testing over on packaging works on a fresh install of Debian 12

**Comment by dhaavi** [comment]
Moved to https://github.com/safing/portmaster-packaging/issues/89.

**Comment by dhaavi** [comment]
Debian 12 does not ship with the "iptables to nftables" translation as they used to. Portmaster still uses iptables for one part of the integration with the network stack.

If you install the "xtables-nft" package, this should do the trick. If anyone comes across this issue and this works, please report back!

We are working on native support for nftables, but as this is a very critical part of the software, it will take some time to build and thoroughly test.

**Comment by kirmaha** [comment]
> If you install the "iptables" package, this should do the trick. If anyone comes across this issue and this works, please report back!

During installation of Portmaster on a fresh Debian 12, using the .deb file, I encountered two errors :
1. Needing to install libnetfilter-queue1 (`sudo apt install libnetfilter-queue1`)
2. Being prompt to manually copy the portmaster.service file (`sudo cp /opt/safing/portmaster/portmaster.service /etc/systemd/system/`). But when doing so, it appeared that the file was already in place. Weird.

Then, starting Portmaster brought two more errors. I didn't write down and forgot the first one, but after login in with my Safing account and then rebooting, it did not show up again. Maybe it is solved, or it was related to the other one. Anyway, the second is what you talked about in this topic :
**Starting module interception failed : Failed to start module: could not initialize nfqueue: exec: "iptables": executable file not found in $PATH**

I installed **iptables** right now (`sudo apt install iptables`) and rebooted. It seems to solve this issue. If the Portmaster is not properly working then, I'll come back here to tell you.
